pkgs/applications/terminal-emulators/hyper/default.nix

```diff
@@ -1,36 +1,58 @@
-{ stdenv, lib, fetchurl, dpkg, atk, glib, pango, gdk-pixbuf, gnome2, gtk3, cairo
-, freetype, fontconfig, dbus, libXi, libXcursor, libXdamage, libXrandr
-, libXcomposite, libXext, libXfixes, libXrender, libX11, libXtst, libXScrnSaver
-, libxcb, nss, nspr, alsa-lib, cups, expat, udev, libpulseaudio, at-spi2-atk }:
+{ stdenv, lib, fetchurl, makeDesktopItem, copyDesktopItems, dpkg, atk, glib, pango
+, gdk-pixbuf, gnome2, gtk3, cairo, freetype, fontconfig, dbus, libXi, libXcursor
+, libXdamage, libXrandr, libXcomposite, libXext, libXfixes, libXrender, libX11, libXtst
+, libXScrnSaver, libxcb, nss, nspr, alsa-lib, cups, expat, udev, libpulseaudio
+, at-spi2-atk, at-spi2-core, libxshmfence, libdrm, libxkbcommon, mesa }:
 
 let
   libPath = lib.makeLibraryPath [
     stdenv.cc.cc gtk3 gnome2.GConf atk glib pango gdk-pixbuf cairo freetype fontconfig dbus
     libXi libXcursor libXdamage libXrandr libXcomposite libXext libXfixes libxcb
     libXrender libX11 libXtst libXScrnSaver nss nspr alsa-lib cups expat udev libpulseaudio
-    at-spi2-atk
+    at-spi2-atk at-spi2-core libxshmfence libdrm libxkbcommon mesa
   ];
+
+  name = "hyper";
+  desktopItem = makeDesktopItem {
+    type = "Application";
+    name = name;
+    desktopName = "Hyper";
+    genericName = "Hyper Terminal";
+    exec = name;
+    icon = "hyper";
+    categories = "System;TerminalEmulator;";
+  };
+
 in
 stdenv.mkDerivation rec {
-  version = "3.0.2";
-  pname = "hyper";
+  pname = name;
+  version = "3.1.1";
+
   src = fetchurl {
-    url = "https://github.com/zeit/hyper/releases/download/${version}/hyper_${version}_amd64.deb";
-    sha256 = "0fv4wv5f8nc739bna83qxmgrvvbyq4w9ch764q2f12wjygrz336p";
+    url = "https://github.com/vercel/hyper/releases/download/v${version}/hyper_${version}_amd64.deb";
+    sha256 = "0j999z649k63rnr1lpbansaxnkrhrp73jdrdmslgnwbh7z1wsp9p";
   };
+  nativeBuildInputs = [ copyDesktopItems ];
+
   buildInputs = [ dpkg ];
+
   unpackPhase = ''
     mkdir pkg
     dpkg-deb -x $src pkg
     sourceRoot=pkg
   '';
+
   installPhase = ''
     mkdir -p "$out/bin"
     mv opt "$out/"
     ln -s "$out/opt/Hyper/hyper" "$out/bin/hyper"
     patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" --set-rpath "${libPath}:$out/opt/Hyper:\$ORIGIN" "$out/opt/Hyper/hyper"
     mv usr/* "$out/"
+    copyDesktopItems ${desktopItem}/share/applications/* $out/share/applications/
   '';
+
+  desktopItems = [ desktopItem ];
```

Comment:
```suggestion
  desktopItems = [
    (makeDesktopItem {
      type = "Application";
      name = "hyper";
      desktopName = "Hyper";
      genericName = "Hyper Terminal";
      exec = "hyper";
      icon = "hyper";
      categories = "System;TerminalEmulator;";
    })
  ];
```

Reply:
I inlined the desktopItem now, but did not store it in a singleton list now, as I copy the desktopItem manually in the installPhase (otherwise we will be left with the defective .desktop file from the deb package.)